task.sql

```diff
@@ -4,30 +4,56 @@ CREATE DATABASE ShopDB;
 USE ShopDB;
 
 CREATE TABLE Countries (
-    ID INT,
+    ID INT AUTO_INCREMENT,
     Name VARCHAR(50),
     PRIMARY KEY (ID)
 );
 
+CREATE TABLE Products (
+    ID INT AUTO_INCREMENT,
+    Name VARCHAR(50),
+    PRIMARY KEY (ID)
+);
+
+CREATE TABLE Warehouses (
+    ID INT AUTO_INCREMENT,
+    Name VARCHAR(50),
+    Address VARCHAR(50),
+    CountryID INT,
+    PRIMARY KEY (ID),
+    FOREIGN KEY (CountryID) REFERENCES Countries(ID) ON DELETE NO ACTION
```

Comment:
The foreign key constraint `ON DELETE NO ACTION` is specified here. Ensure that this is the intended behavior. If you want to prevent deletion of a country if it's referenced in the `Warehouses` table, this is correct. Otherwise, consider using `ON DELETE CASCADE` or `ON DELETE SET NULL` depending on your requirements.